lib/irb/ruby-lex.rb

```diff
@@ -30,26 +30,25 @@ def initialize
     @prompt = nil
   end
 
-  def self.compile_with_errors_suppressed(code)
-    line_no = 1
+  def self.compile_with_errors_suppressed(code, line_no: 1)
     begin
       result = yield code, line_no
     rescue ArgumentError
       code = ";\n#{code}"
-      line_no = 0
+      line_no -= 1
```

Comment:
Minor: Perhaps a code comment here, to explain the meaning of the negative number?

Reply:
I didn't understand why this rescue existed, but with some help from git blame, I found it is due to incomplete magic comment handling.  I pushed another commit explaining not just the line number issue, but the point of the rescue block in general.